### PR TITLE
[REVIEW] libxx Cython reorganization

### PR DESCRIPTION
**Pull request by shwina**
* Rename `_libxx.includes` to `_libxx.cpp`. *Only* `.pxd` files containing `extern` declarations go here.
* Organize `.pxd` files in `_libxx.cpp` in a similar way to libcudf headers.
* Create a `move.pxd` file containing all `move` declarations.
* Remove `lib.pxd`, splitting it into appropriate `.pxd` files in `_libxx/cpp`.
* Eliminate all uses of `import *` and `cimport *` in `_libxx/`

### CODE REVIEW

File: python/cudf/cudf/_libxx/aggregation.pyx [hunk 1]
```diff
@@ -5,12 +5,21 @@ import numba
...
+from cudf._libxx.move cimport move
+
+from cudf._libxx.aggregation cimport get_aggregation
+from cudf._libxx.cpp.types cimport (
+    type_id,
+    size_type,
+    data_type,
+    interpolation
+)
+cimport cudf._libxx.cpp.aggregation as cudf_aggregation
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/aggregation.pyx]: This feels like an awkward naming convention.

Comment (kkraus14) [reply python/cudf/cudf/_libxx/aggregation.pyx]:
```suggestion
cimport cudf._libxx.cpp.aggregation as cpp_aggregation
```

Comment (shwina) [reply python/cudf/cudf/_libxx/aggregation.pyx]: Agreeg. But also everywhere else that we do

`cimport cudf._libxx.cpp.types as cudf_types`

should we do instead:

`cimport cudf._libxx.cpp.types as cpp_types`

I think that can be confusing

Comment (kkraus14) [reply python/cudf/cudf/_libxx/aggregation.pyx]: Naming is hard 😞, I think we should just pick a prefix and stick with it for all externed functions / modules. I'm fine with `cpp_*` or `libcudf_*` but `cudf_*` is super ambiguous.

Comment (shwina) [reply python/cudf/cudf/_libxx/aggregation.pyx]: `cpp_` is even more so. `libcudf_` makes the most sense to me

File: python/cudf/cudf/_libxx/cpp/column/column.pxd [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from libcpp.memory cimport unique_ptr
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/cpp/column/column.pxd]: Copyright

File: python/cudf/cudf/_libxx/cpp/column/column_view.pxd [hunk 3]
```diff
@@ -0,0 +1,122 @@
+from libcpp.vector cimport vector
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/cpp/column/column_view.pxd]: Copyright

File: python/cudf/cudf/_libxx/cpp/dlpack.pxd [hunk 4]
```diff
@@ -5,9 +5,12 @@
 # cython: embedsignature = True
 # cython: language_level = 3
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/cpp/dlpack.pxd]: Remove this block

File: python/cudf/cudf/_libxx/cpp/join.pxd [hunk 5]
```diff
@@ -0,0 +1,30 @@
+from libcpp.memory cimport unique_ptr
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/cpp/join.pxd]: Copyright

File: python/cudf/cudf/_libxx/cpp/merge.pxd [hunk 6]
```diff
@@ -0,0 +1,15 @@
+from libcpp.vector cimport vector
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/cpp/merge.pxd]: Copyright

File: python/cudf/cudf/_libxx/cpp/quantiles.pxd [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/cpp/quantiles.pxd]: Remove this block

File: python/cudf/cudf/_libxx/cpp/scalar/scalar.pxd [hunk 8]
```diff
@@ -0,0 +1,44 @@
+from libc.stdint cimport (
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/cpp/scalar/scalar.pxd]: Copyright

File: python/cudf/cudf/_libxx/cpp/search.pxd [hunk 9]
```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+from libcpp.vector cimport vector
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/cpp/search.pxd]:
```suggestion

from libcpp.vector cimport vector
```

File: python/cudf/cudf/_libxx/cpp/sort.pxd [hunk 10]
```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+from libcpp.memory cimport unique_ptr
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/cpp/sort.pxd]:
```suggestion

from libcpp.memory cimport unique_ptr
```

File: python/cudf/cudf/_libxx/cpp/table/table.pxd [hunk 11]
```diff
@@ -0,0 +1,19 @@
+from libcpp.vector cimport vector
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/cpp/table/table.pxd]: Copyright

File: python/cudf/cudf/_libxx/cpp/table/table_view.pxd [hunk 12]
```diff
@@ -0,0 +1,22 @@
+from libcpp.vector cimport vector
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/cpp/table/table_view.pxd]: Copyright

File: python/cudf/cudf/_libxx/cpp/types.pxd [hunk 13]
```diff
@@ -0,0 +1,59 @@
+from libc.stdint cimport int32_t, uint32_t
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/cpp/types.pxd]: Copyright

File: python/cudf/cudf/_libxx/cpp/wrappers/bool.pxd [hunk 14]
```diff
@@ -0,0 +1,4 @@
+from libcpp cimport bool
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/cpp/wrappers/bool.pxd]: Copyright

File: python/cudf/cudf/_libxx/cpp/wrappers/timestamps.pxd [hunk 15]
```diff
@@ -0,0 +1,8 @@
+from libc.stdint cimport int64_t
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/cpp/wrappers/timestamps.pxd]: Copyright

File: python/cudf/cudf/_libxx/dlpack.pyx [hunk 16]
```diff
@@ -7,15 +7,18 @@
 
 # Copyright (c) 2018, NVIDIA CORPORATION.
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/dlpack.pyx]:
```suggestion
```

File: python/cudf/cudf/_libxx/dlpack.pyx [hunk 17]
```diff
@@ -7,15 +7,18 @@
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/dlpack.pyx]: Remove block above this

File: python/cudf/cudf/_libxx/join.pxd [hunk 18]
```diff
@@ -2,22 +2,23 @@ from libcpp.vector cimport vector
 from libcpp.pair cimport pair
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/join.pxd]: Copyright

File: python/cudf/cudf/_libxx/join.pyx [hunk 19]
```diff
@@ -1,10 +1,17 @@
-from cudf._libxx.lib cimport *
-from cudf._libxx.table cimport Table, columns_from_ptr
-from cudf._libxx.includes.join cimport *
+from collections import OrderedDict
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/join.pyx]: Copyright

File: python/cudf/cudf/_libxx/merge.pyx [hunk 20]
```diff
@@ -1,8 +1,15 @@
-from cudf._libxx.lib cimport *
+from libcpp.vector cimport vector
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/merge.pyx]: Copyright

File: python/cudf/cudf/_libxx/move.pxd [hunk 21]
```diff
@@ -0,0 +1,49 @@
+from libcpp.memory cimport unique_ptr
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/move.pxd]: Copyright

File: python/cudf/cudf/_libxx/types.pxd [hunk 22]
```diff
@@ -0,0 +1,8 @@
+from libcpp cimport bool
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/types.pxd]: Copyright

**Pull request merged**